docs/source/features/memory_optimizations.rst

```diff
@@ -11,14 +11,16 @@ Flash Attention
+
+The flash attention algorithm was first propsed `here <https://arxiv.org/pdf/2205.14135>`_, and two of its implementations are `flash-attention <https://github.com/Dao-AILab/flash-attention>`_ by Tri Dao *et al*, and `fused flash attention <https://docs.nvidia.com/deeplearning/cudnn/archives/cudnn-897/developer-guide/index.html#flash-fused-multi-head-att-fprop>`_ by NVIDIA cuDNN.
 
 Turn Flash Attention On and Off
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-In the NeMo Framework, Flash Attention is supported through the Transformer Engine with the inclusion of Flash Attention 2. By default, Flash Attention is enabled, but the Transformer Engine may switch to a different kernel if the tensor dimensions are not optimal for Flash Attention. Users can completely disable Flash Attention by setting the environment variable ``NVTE_FLASH_ATTN=0``.
+In the NeMo Framework, flash attention is supported through `Transformer Engine <https://github.com/NVIDIA/TransformerEngine/tree/main>`_ with both of the above implementations. Transformer Engine selects the appropriate implementation based on the input information (sequence length, number of heads, head dimension, etc), but when both implementations are applicable, Transformer Engine prefers cuDNN flash attention on Hopper+ architectures, and Tri Dao flash attention on Ampere-based architectures. To disable Tri Dao flash attention, users can set the environment variable ``NVTE_FLASH_ATTN=0``, and to disable cuDNN flash attention, users can set ``NVTE_FUSED_ATTN=0``.
 
-For more details on the supported Dot Attention backend, please refer to the Transformer Engine source code available at `Transformer Engine's Attention Mechanism <https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/attention.py>`_.
+For more details on the Dot Product Attention backends supported in Transformer Engine, please refer to the source code at `Transformer Engine's Attention Mechanism <https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/attention.py>`_.
 
 Activation Recomputation
 ------------------------
```

Comment:
Comment on #33 Change "This method recalculates . . ." to "The full activation recomputation method recalculates . . ."

Comment on #37 Change "This method recalculates . . ." to "The partial activation recomputation method recalculates . . ."

Comment on #41 Change "This method recalculates . . ." to "The selective activation recomputation method recalculates . . ."